File: Jack.Business/Calcado.vb
Imports Jack.Extensions.Basic
Public Class Calcado

    Implements ICrud(Of Model.Calcado, Integer)

    Public Sub New()

    End Sub

    Public Function Delete(oTipo As Model.Calcado) As Boolean Implements ICrud(Of Model.Calcado, Integer).Delete
        Dim oDados As Data.Calcado = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Calcado
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.Calcado Implements ICrud(Of Model.Calcado, Integer).Find

        Dim oDados As Data.Calcado = Nothing
        Dim lstRetorno As IList(Of Model.Calcado) = Nothing

        Try
            oDados = New Data.Calcado
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Insert(oTipo As Model.Calcado) As Boolean Implements ICrud(Of Model.Calcado, Integer).Insert
        Dim oDados As Data.Calcado = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Calcado
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function LoadAll() As IList(Of Model.Calcado) Implements ICrud(Of Model.Calcado, Integer).LoadAll
        Dim oDados As Data.Calcado = Nothing
        Dim lstRetorno As IList(Of Model.Calcado) = Nothing

        Try
            oDados = New Data.Calcado
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Update(oTipo As Model.Calcado) As Boolean Implements ICrud(Of Model.Calcado, Integer).Update
        Dim oDados As Data.Calcado = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Calcado
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
End Class

File: Jack.Business/Calcado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Jack;

namespace Jack.Business
{
    public class Calcado : ICrud<Model.Calcado, int>
    {


        public Calcado()
        {
        }

        public bool Delete(Model.Calcado oTipo)
        {
            Data.Calcado oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Calcado();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.Calcado Find(int Identifier)
        {

            Data.Calcado oDados = null;
            IList<Model.Calcado> lstRetorno = null;

            try
            {
                oDados = new Data.Calcado();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Insert(Model.Calcado oTipo)
        {
            Data.Calcado oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Calcado();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.Calcado> LoadAll()
        {
            Data.Calcado oDados = null;
            IList<Model.Calcado> lstRetorno = null;

            try
            {
                oDados = new Data.Calcado();
                //lstRetorno = oDados.LoadAll().OrderByDescending(Function(x) x.MedidaIdade).OrderBy(Function(x) x.Sexo).OrderBy(Function(x) x.IdadeInicial).ToList()
                lstRetorno = (from oDado in oDados.LoadAll() orderby oDado.MedidaIdade descending, oDado.Sexo, oDado.IdadeInicial).ToList();


            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Update(Model.Calcado oTipo)
        {
            Data.Calcado oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Calcado();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
    }
}